test/redis/cache_impl_test.go

```diff
@@ -189,6 +227,57 @@ func TestOverLimitWithLocalCache(t *testing.T) {
 	assert.Equal(uint64(0), limits[0].Stats.OverLimitWithLocalCache.Value())
 	assert.Equal(uint64(0), limits[0].Stats.NearLimit.Value())
 
+	// Check the local cache stats.
+	localCacheStats.GenerateStats()
+	statsStore.Flush()
+
+	pattern :=
+		`localcache.averageAccessTime:\d+|g\n` +
+			`localcache.hitCount:\d+|g\n` +
+			`localcache.missCount:\d+|g\n` +
+			`localcache.lookupCount:\d+|g\n` +
+			`localcache.overwriteCount:\d+|g\n` +
+			`localcache.evacuateCount:\d+|g\n` +
+			`localcache.expiredCount:\d+|g\n` +
+			`localcache.entryCount:\d+|g\n`
+
+	// Check whether all local_cache related stats are available.
+	ok, err := regexp.Match(pattern, []byte(sink.record))
+	if err != nil {
+		t.Fatal("Error: ", err)
+	}
+	if !ok {
+		t.Errorf("Expected: '%s' Got: '%s'", pattern, sink.record)
+	}
+
+	// Check the correctness of hitCount, missCount, lookupCount, expiredCount and entryCount
+	submatch := regexp.MustCompile(`localcache.hitCount:(\d+)\|g\n`).FindStringSubmatch(sink.record)
```

Comment:
There is a huge amount of duplication between each test on how you deal with the regex, etc. Can you unify that into a utility? Also, at a higher layer, testing the stats in this way seems suboptimal. Can't you just store the stats as integers vs. strings and then look at them directly?

Reply:
Just moved it to the utility method, and also simplified it.